python/instrumentation/openinference-instrumentation-bedrock/src/openinference/instrumentation/bedrock/__init__.py

```diff
@@ -418,3 +391,53 @@ def _get_attributes_from_image(
 
 def is_iterable_of(lst: Iterable[object], tp: T) -> bool:
     return isinstance(lst, Iterable) and all(isinstance(x, tp) for x in lst)
+
+
+def set_model_name(span: Span, response_body: Dict[str, Any], kwargs: Dict[str, Any]) -> None:
+    if model_id := kwargs.get("modelId"):
+        _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+        vendor = None
+        if isinstance(model_id, str):
+            (vendor, *_) = model_id.split(".")
+
+        if vendor == "ai21":
+            content = str(response_body.get("completions"))
+        elif vendor == "anthropic":
+            content = str(response_body.get("completion"))
+        elif vendor == "cohere":
+            content = str(response_body.get("generations"))
+        elif vendor == "meta":
+            content = str(response_body.get("generation"))
+        else:
+            content = ""
```

Comment:
This feels brittle, why not take a more feature detection approach of checking the body contents? Tying it to vendors seems non-forward thinking